agent/src/containeroptions/containeroptionparser.cpp

```diff
@@ -36,14 +37,35 @@ void ContainerOptionParser::readConfigElement(const json_t *element)
         throw ContainerOptionParseError(errorMessage);
     }
 
-    bool wo = false;
-    if(!JSONParser::read(element, "enableWriteBuffer", wo)) {
+    bool enableWriteBuffer = false;
+    if(!JSONParser::read(element, "enableWriteBuffer", enableWriteBuffer)) {
         std::string errorMessage("Could not parse config due to: 'enableWriteBuffer' not found.");
         log_error() << errorMessage;
         throw ContainerOptionParseError(errorMessage);
     }
 
-    m_options->setEnableWriteBuffer(wo);
+    m_options->setEnableWriteBuffer(enableWriteBuffer);
+
+    if (enableWriteBuffer == true) {
+        bool enableTemporaryFilesystemWriteBuffer = false;
+        if(!JSONParser::read(element,
+                             "enableTemporaryFileSystemWriteBuffer",
+                             enableTemporaryFilesystemWriteBuffer)) {
```

Comment:
The '{' goes on the next line when the if is broken up like this, as I mentioned in previous review.